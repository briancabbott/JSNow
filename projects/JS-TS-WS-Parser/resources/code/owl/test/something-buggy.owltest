---program = decl*
decl =
 'class' identifier [ '{' class-body '}' ] : class
 'function' identifier [ '(' ((param ',')* param)? ')' ] [ '{' stmt* '}' ] : function
class-body = class-decl*
class-decl =
  'method' identifier [ '(' ((param ',')* param)? ')' ] [ '{' stmt* '}' ] : method
  'ivar' identifier : ivar
param = identifier
stmt =
  expr [ '(' ((expr ',')* expr)? ')' ] : call
  'var' identifier ('=' expr)? : var
  identifier '=' expr : assignment
  'if' expr@if-expr [ '{' stmt@if-stmt* '}' ]
   ('else' [ '{' stmt@else-stmt* '}' ])? : if
  'while' expr [ '{' stmt* '}' ] : while
expr =
  identifier : ident
  number : num
  string : string
  'true' : true
  'false' : false
  [ '(' expr ')' ] : parens
 .operators postfix
  [ '(' ((expr ',')* expr)? ')' ] : call
 .operators prefix
  '-' : negate
 .operators infix flat 
  '*' : times
  '/' : divide
 .operators infix flat
  '+' : plus
  '-' : minus
 .operators infix nonassoc
  '==' : eq
  '>=' : ge
  '<=' : le
  '>' : gt
  '<' : lt
