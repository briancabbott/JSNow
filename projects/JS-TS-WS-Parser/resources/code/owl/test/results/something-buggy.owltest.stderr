error: this grammar is ambiguous

. function a ( ) { "a" ( ) - false ( ) } 

  can be parsed in two different ways: as

. function a ( ) { "a"         ( ) - false      ( ) } 
  |                expr:string       |                
  |                expr:call------   expr:false       
  |                expr:minus------------------       
  |                stmt:call-----------------------   
  decl:function-------------------------------------- 
  program-------------------------------------------- 

  or as

. function a ( ) { "a"         ( ) - false      ( ) } 
  |                |               | expr:false       
  |                expr:string     expr:negate-       
  |                stmt:call------ stmt:call-------   
  decl:function-------------------------------------- 
  program-------------------------------------------- 

